### DISCUSSION TITLE
[New Feature] Cover for infantry

### DISCUSSION THREAD

**Discussion by CCTiberius** [Vinifera-Developers/Vinifera#1508] [Feature Requests]
### Description:
A common complaint I see is that "urban combat" isn't very consequential in TS. Allowing infantry to somehow take cover behind buildings and rubble could go a long way. Could also apply to certain terrain.

### Possible Implementation:
If an infantry is standing behind a building, it takes X% less damage. Maybe it'd even be possible to add a stage to infantry like prone state, but "cover state" so if they are near a building or some rubble or big rocks, they go into "cover" and take X% less dmg from all sources.

**Comment by Crimsonum** [comment]
As a simpler and more general implementation – although these don't have to be mutually exclusive – I would add a key that makes the projectile subject to obstacles like buildings. YR already has keys that achieve a similar effect (e.g. `SubjectToWalls`). Regardless of whether the target is infantry, vehicle or building, you shouldn't be able to fire a machine gun through a skyscraper and expect to hit things behind it. This is already how arcing projectiles behave, but invisible and homing projectiles ignore obstacles in their path and simply fly through them.

**Comment by CCTiberius** [comment]
I think having both would be nice yeah
